### PR TITLE
Add dependency chapter

### PR DESCRIPTION
**Pull request by uncoverd**
Task: [#300](https://app.productive.io/1-infinum/tasks/1048120)

#### Aim
Add a chapter about dependencies. It should answer the following questions:

- are dependencies really a big deal?
- why should I bother updating dependencies?
- what tools can I use for managing updates?
- how can I update dependencies safely?
- are there any best practices for Rails upgrades?

### CODE REVIEW

**Review by melcha — COMMENTED** [review]
Nicely written, a lot of useful information and guidelines for updating dependencies.

File: Workflows/Upgrading.md [hunk 1]
```diff
@@ -1,4 +1,121 @@
...
+
+- All dependencies have a certain support life cycle. If you're using an older version and a new crucial security update is announced you may be forced to do a rapid upgrade of multiple versions to secure your application. Doing that is risky, especially in a time crunch.
+-  Regular smaller dependency upgrades are more predictable, safer and easier to manage in the context of a sprint. Failing to update dependencies regularly can turn the upgrade process from several bite size chunks which can be done independently to a massive undertaking that takes days and blocks all development. Estimating these big upgrades is always hard and we usually underestimate it by a factor of 2 or 3.
```
Comment (d4be4st) [inline Workflows/Upgrading.md:24]: or 10 :D

**Review by d4be4st — APPROVED** [review]

**Review by nikajukic — APPROVED** [review]
Great job @uncoverd! 👏

**Review by vr4b4c — APPROVED** [review]
Great article! 🤘 👏

**Review by melcha — APPROVED** [review]
🎉 🎉 🎉 
This could be a blog post.

**Pull request merged**